xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -439,8 +383,9 @@ void CPeripheralBusAddon::UpdateAddons(void)
   std::set<std::string> removed;
 
   // Get new add-ons
-  PeripheralAddonVector newAddons = GetEnabledAddons();
-  std::transform(newAddons.begin(), newAddons.end(), std::inserter(newIds, newIds.end()), GetPeripheralAddonID);
+  BinaryAddonBaseList newAddons;
+  CServiceBroker::GetBinaryAddonManager().GetAddonInfos(newAddons, true, ADDON_PERIPHERALDLL);
```

Comment:
Like here. This doesn't work. CPeripheralBusAddon listens to events from addon manager, so there is no guarantee BinaryAddonManager will have the correct list at this point. And then we are out of sync.